client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx

```diff
@@ -0,0 +1,114 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { FunctionComponent, useContext, useCallback } from 'react';
+import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
+import { getPurchaseURLCallback } from 'calypso/my-sites/plans/jetpack-plans/get-purchase-url-callback';
+import slugToSelectorProduct from 'calypso/my-sites/plans/jetpack-plans/slug-to-selector-product';
+import { SelectorProduct } from 'calypso/my-sites/plans/jetpack-plans/types';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+
+import './style.scss';
+
+type Props = {
+	productSlug: string;
+	onClose: () => void;
+	siteId?: number;
+	currentProduct: APIProductFamilyProduct;
+};
+
+const LicenseLightboxPurchaseViaJetpackcom: FunctionComponent< Props > = ( {
+	productSlug,
+	onClose,
+	siteId,
+	currentProduct,
+} ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const { hideLicenseInfo } = useContext( SitesOverviewContext );
+
+	const item = slugToSelectorProduct( productSlug ) as SelectorProduct;
+
+	const selectedSiteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) ) || '';
+
+	const createCheckoutURL = getPurchaseURLCallback( selectedSiteSlug, {
+		// For the Backup, Search, Scan upsell in Jetpack Cloud, we want to redirect back here to the Backup page after checkout.
+		redirect_to: window.location.href,
+	} );
+
+	const checkoutURL = createCheckoutURL && createCheckoutURL( item, false );
+	const onHideLicenseInfo = useCallback( () => {
+		hideLicenseInfo();
+		onClose?.();
+	}, [ hideLicenseInfo, onClose ] );
+
+	const onProceedToCheckout = useCallback( () => {
+		dispatch(
+			recordTracksEvent( 'calypso_jetpack_single_site_upsell_proceed_to_checkout_click', {
+				product: currentProduct.slug,
+			} )
+		);
+		onHideLicenseInfo();
+	}, [ currentProduct, dispatch, onHideLicenseInfo ] );
+
+	const monthlyPrice = parseFloat( currentProduct.amount );
+
+	let yearlyPrice = 0;
+
+	if ( currentProduct.currency === 'USD' ) {
+		yearlyPrice = monthlyPrice * 12;
+	}
+
+	const learnMoreLink =
+		'https://jetpack.com/support/jetpack-manage-instructions/jetpack-manage-billing-payment-faqs';
```

Comment:
Probably should use `localizeURL()` on this.

Reply:
👍 Thanks for mentioning that, added now.